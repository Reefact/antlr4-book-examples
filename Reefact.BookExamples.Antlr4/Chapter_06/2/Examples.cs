#region Usings declarations

using ApprovalTests;
using ApprovalTests.Reporters;

using Xunit;

#endregion

namespace Reefact.BookExamples.Antlr4.Chapter_06._2 {

    [UseReporter(typeof(VisualStudioReporter))]
    public class Examples {

        [Fact]
        public void t_json() {
            // Setup
            GRun grun = GRun.ReadResource("t.json", 6, 2);
            // Exercise
            string tokensString = grun.ToTokensString();
            // Verify
            Approvals.Verify(tokensString);
        }

        [Fact]
        public void simple_array_tokens() {
            // Setup
            GRun grun = GRun.ReadString("[1,\"\\u0049\",1.3e9]");
            // Exercise
            string tokensString = grun.ToTokensString();
            // Verify
            Approvals.Verify(tokensString);
        }

        [Fact]
        public void simple_array_tree() {
            // Setup
            GRun grun = GRun.ReadString("[1,\"\\u0049\",1.3e9]");
            // Exercise
            string lispStyleTree = grun.ToLispStyleTree();
            // Verify
            Approvals.Verify(lispStyleTree);
        }

    }

}