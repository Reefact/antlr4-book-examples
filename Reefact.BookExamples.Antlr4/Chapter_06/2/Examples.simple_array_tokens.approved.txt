[@0,0:0='[',<5>,1:0]
[@1,1:1='1',<11>,1:1]
[@2,2:2=',',<2>,1:2]
[@3,3:10='"\u0049"',<10>,1:3]
[@4,11:11=',',<2>,1:11]
[@5,12:16='1.3e9',<11>,1:12]
[@6,17:17=']',<6>,1:17]
[@7,18:17='<EOF>',<-1>,1:18]